resource_tracker/models/resource_group_attribute_definition.py

```diff
@@ -30,20 +31,27 @@ class ResourceGroupAttributeDefinition(models.Model):
     def __str__(self):
         return f"{self.resource_group} - {self.name}"
 
-    def calculate_total_resource(self):
-        total = 0
-        for resource in self.resource_group.resources.all():
-            try:
-                total += resource.attributes.get(attribute_type=self).value
-            except ResourceAttribute.DoesNotExist:
-                pass
-        self.total_resource = total
+    def calculate_total_resource(self, delta=None):
+        if delta is None:
+            total = 0
+            for resource in self.resource_group.resources.all():
+                try:
+                    total += resource.attributes.get(attribute_type=self).value
+                except ResourceAttribute.DoesNotExist:
+                    pass
+            self.total_resource = total
+            # sync pool attributes
+            if self.consume_from is not None:
+                self.consume_from.calculate_total_consumed()
+            if self.produce_for is not None:
+                self.produce_for.calculate_total_produced()
```

Comment:
On this part:
- identify when we send delta=none
- place this part in a celery task
- return the task ID to the controller